examples/server/server.cpp

```diff
@@ -86,6 +97,40 @@ static void server_log(const char * level, const char * function, int line,
     fflush(stdout);
 }
 
+// format incomplete utf-8 multibyte character for output
+static std::string tokens_to_output_formatted_string(const llama_context * ctx, const llama_token token) {
+    std::string out = token == -1 ? "" : llama_token_to_str(ctx, token);
+    // if first bit is 1, meaning it's a partial character
+    if ((out[0] & 0x80) == 0x80) {
+        std::stringstream ss;
+        ss<< std::hex << (out[0] & 0xff);
+        std::string res ( ss.str() );
+        out = "byte: \\x" + res;
+    }
+    return out;
+}
```

Comment:
This function seems kind of hacky. What format is `byte: \x00` supposed to be?

It also seems incorrect as it only looks at the first character, but that doesn't mean that the continuation bytes are not there.

Maybe for the first version we can just let the JSON library replace invalid sequences with �

Reply:
based on https://huggingface.co/oobabooga/llama-tokenizer/raw/main/tokenizer.json anon provided, all incomplete tokens are one byte long. So if `out` is a partial char, it should be exactly one byte and starts with 0b1.

Reply:
The `byte: \x00` is for easy reading and printing on the client side. Based on the syntax [here](https://design215.com/toolbox/ascii-utf8.php#:~:text=UTF%2D8%202%2Dbyte%20Characters%3A%20byte%201%20%3D%20%5Cxc0%2D%5Cxdf%2C%20byte%202%20%3D%20%5Cx80%2D%5Cxbf)

Reply:
Well, it's not easy because it's supposed to be machine readable. Maybe an option is to encode it in a different JSON property as a number value?